### PR TITLE
Feature/user rest controller

### PR DESCRIPTION
**Pull request by brunovt074**
Added an UserController and UserDTO, along with partially implemented methods on UserService, UserServiceImpl and UserController. ExceptionHandling and Testing are pending.
If everything is ok I could proceed with User Entity and the Exception Handling.

### CODE REVIEW

**Review by MarkoGajic93 — APPROVED** [review]

**Review by lardezzoni — APPROVED** [review]
Ok

**Review by savicsoft — APPROVED** [review]
We want the base url to be localhost:8080/api/v1/users

File: src/main/java/com/savicsoft/carpooling/user/dto/UserDTO.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Value
+@Builder
+public class UserDTO {
+    Long id;
+
+    public static UserDTO fromUser(User user){
+        return UserDTO.builder()
+                .id(user.getId())
+                .build();
+    }
```
Comment (sergienkoyura) [inline src/main/java/com/savicsoft/carpooling/user/dto/UserDTO.java:19]: We were talking about MapStruct as a Mapper so I guess you can implement it

Comment (brunovt074) [reply src/main/java/com/savicsoft/carpooling/user/dto/UserDTO.java:19]: Ah, I didn't know that. So, should I remove the @Builder annotation, the constructor and add a mapper??

**Pull request closed without merging**